### PR TITLE
Remove unnecessary calls to Presenter's onDestroy

### PR DESCRIPTION
**Pull request by Boza-s6**
Presenter#onDestory would be called three time in a row, from
Loader's onReset, PresenterActivity's onLoaderReset and onDestory.

Removed calls from PresenterActivity.

### CODE REVIEW

File: app/src/main/kotlin/io/github/plastix/kotlinboilerplate/ui/base/PresenterActivity.kt [hunk 1]
```diff
@@ -27,11 +27,6 @@ abstract class PresenterActivity<V : MvpView, T : Presenter<V>> : BaseActivity()
     }
 
     @CallSuper
-    protected fun onPresenterDestroyed() {
```
Comment (Plastix) [inline app/src/main/kotlin/io/github/plastix/kotlinboilerplate/ui/base/PresenterActivity.kt:30]: This is a hook for subclasses to override if they want to perform an action when the presenter gets destroyed.

**Comment by Plastix** [comment]
Both the Activity's `onDestroy` method and Loader callback `onLoaderReset` delegate to `onPresenterDestroyed` which calls the presenters destroy method. Thus, it really is only being called _twice_ not three times.

Really we shouldn't be calling the presenter's destroy method from the activity `onDestroy` and just doing it in `onLoaderReset`.

**Comment by Boza-s6** [comment]
It's called from PresenterLoader `onReset` mehtod, also.

File: app/src/main/kotlin/io/github/plastix/kotlinboilerplate/ui/base/PresenterActivity.kt (hunk 1 above)
Comment (Boza-s6) [reply app/src/main/kotlin/io/github/plastix/kotlinboilerplate/ui/base/PresenterActivity.kt:30]: Mhm, ok. I'll leave method in place.

**Comment by Plastix** [comment]
Thoughts? https://github.com/Plastix/Kotlin-Android-Boilerplate/commit/5bef3557c9fee2983d7b2fda7e61d9c6fde96fb3

**Comment by Boza-s6** [comment]
You'll still have 2 calls onDestory, because it's called from Loader too. Check this [PresenterLoader](https://github.com/Plastix/Kotlin-Android-Boilerplate/blob/master/app/src/main/kotlin/io/github/plastix/kotlinboilerplate/ui/base/PresenterLoader.kt#L32).

I think you should remove `presenter.onDestroy()` from `onPresenterDestroyed`.

**Comment by Plastix** [comment]
Oh good catch. Thanks

**Comment by Plastix** [comment]
Closed with https://github.com/Plastix/Kotlin-Android-Boilerplate/commit/5d3bc5120953efd10559fda12ae5dab16a72f6bb 👍  Really appreciate the code review.

**Pull request closed without merging**